src/View/Helper/BreadcrumbsHelper.php

```diff
@@ -0,0 +1,295 @@
+        ]
+    ];
+
+    /**
+     * The crumbs list.
+     *
+     * @var array
+     */
+    protected $crumbs = [];
+
+    /**
+     * Add a crumb to the trail.
+     *
+     * @param string|array $title Title of the crumb. If provided as an array, it will add each values of the array
+     * as a single crumb. This allows you to add multiple crumbs in the trail at once. Arrays are expected to be of this
+     * form:
+     * - *title* The title of the crumb
+     * - *link* The link of the crumb. If not provided, no link will be made
+     * - *options* Options of the crumb. See description of params option of this method.
+     * @param string|array|null $url URL of the crumb. Either a string, an array of route params to pass to
+     * Url::build() or null / empty if the crumb does not have a link
+     * @param array $options Array of options. These options will be used as attributes HTML attribute the crumb will
+     * be rendered in (a <li> tag by default). It accepts two special keys:
+     * - *innerAttrs*: An array that allows you to define attributes for the inner element of the crumb (by default, to
+     * the link)
+     * - *templateVars*: Specific template vars in case you override the templates provided
+     * @return $this
+     */
+    public function add($title, $url = null, array $options = [])
+    {
+        if (is_array($title)) {
+            foreach ($title as $crumb) {
+                $this->crumbs[] = $crumb + ['title' => '', 'url' => null, 'options' => []];
+            }
+
+            return $this;
+        }
+
+        $this->crumbs[] = compact('title', 'url', 'options');
+
+        return $this;
+    }
+
+    /**
+     * Prepend a crumb to the start of the queue.
+     *
+     * @param string $title Title of the crumb
+     * @param string|array|null $url URL of the crumb. Either a string, an array of route params to pass to
+     * Url::build() or null / empty if the crumb does not have a link
+     * @param array $options Array of options. These options will be used as attributes HTML attribute the crumb will
+     * be rendered in (a <li> tag by default). It accepts two special keys:
+     * - *innerAttrs*: An array that allows you to define attributes for the inner element of the crumb (by default, to
+     * the link)
+     * - *templateVars*: Specific template vars in case you override the templates provided
+     * @return $this
+     */
+    public function prepend($title, $url = null, array $options = [])
+    {
+        array_unshift($this->crumbs, compact('title', 'url', 'options'));
+
+        return $this;
+    }
+
+    /**
+     * Insert a crumb at a specific index.
+     *
+     * If the index already exists, the new crumb will be inserted,
+     * and the existing element will be shifted one index greater.
+     *
+     * @param int $index The index to insert at.
+     * @param string $title Title of the crumb
```

Comment:
Do we have a standard settled for this ?
It would good to have a sniff for that if so. This is typically the kind of thing I absolutely don't pay attention to.